File: 509-fibonacci-number/509-fibonacci-number.py
class Solution(object):
    def fib(self, n):
        if n==0:
            return 0
        if n==1:
            return 1
        fib = [0,1]
        for i in range(2,n):
            fib.append(fib[-1] + fib[-2])
        return fib[-1] + fib[-2]

File: 509-fibonacci-number/509-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        int pp=0;
        int p=1;
        if(n==0) return 0;
        if(n==1) return 1;
        for(int i=2;i<=n;i++){
            int curr = pp + p;
            pp = p;
            p = curr;
        }
        return p;
    }
};